path_data.go

```diff
@@ -18,11 +18,15 @@ import (
 	"github.com/mitchellh/mapstructure"
 )
 
+func matchAllNoTrailingSlashRegex(name string) string {
+	return fmt.Sprintf(`(?P<%s>.*?[^/])`, name)
```

Comment:
I think in most cases this might just match 1 character after `data/`? i.e. the `.*?` captures as little as it can until it finds a non `/` character, which in most cases will be the first character. I could definitely be reading it wrong, but would be good to have some tests to confirm.

Reply:
@tomhjp great shout, thanks for this Tom.

I should have added tests before I moved the PR out of draft, my bad. I've tweaked the regex now and added some tests which hopefully reflect the logic we're trying to check.

:)